File: Daily Test/C++/Minimum - Maximum In Subarrays 14.1.2026.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, k;
    cin >> n >> k;

    vector<int> arr(n);
    for (int i = 0; i < n; i++)
        cin >> arr[i];

    int ans = INT_MAX;

    for (int i = 0; i <= n - k; i++) {
        int m = *max_element(arr.begin() + i, arr.begin() + i + k);
        ans = min(ans, m);
    }

    cout << ans;
    return 0;
}

File: Daily Test/Java/Minimum - Maximum In Subarrays 14.1.2026.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int k = sc.nextInt();

        int[] arr = new int[n];
        for (int i = 0; i < n; i++)
            arr[i] = sc.nextInt();

        int ans = Integer.MAX_VALUE;

        for (int i = 0; i <= n - k; i++) {
            int m = arr[i];
            for (int j = i; j < i + k; j++) {
                if (arr[j] > m)
                    m = arr[j];
            }
            ans = Math.min(ans, m);
        }

        System.out.println(ans);
    }
}